src/common/util.cpp

```diff
@@ -1077,13 +1077,25 @@ std::string get_nix_version_display_string()
     if (seconds < 3600)
       return std::to_string((uint64_t)(seconds / 60)) + " minutes";
     if (seconds < 3600 * 24)
-      return std::to_string((uint64_t)(seconds / 3600)) + " hours";
+    {
+      uint64_t full_hours = seconds / 3600;
+      uint64_t tenths = (seconds % 3600) / (3600 / 10);
+      return std::to_string(full_hours) + "." + std::to_string(tenths) + " hours";
+    }
     if (seconds < 3600 * 24 * 30.5)
-      return std::to_string((uint64_t)(seconds / (3600 * 24))) + " days";
+    {
+      uint64_t full_days = seconds / (3600 * 24);
+      if (full_days > 5)
+      {
+        return std::to_string(full_days) + " days";
+      }
+      uint64_t tenths = seconds % (3600 * 24) / (3600 * 24 / 10);
+      return std::to_string(full_days) + "." + std::to_string(tenths) + " days";
+    }
```

Comment:
It'd seem much simpler to either change the unit switch threshold to ten times what they are, or print a float with one fractional digit rather than the custom extra parts above.

Reply:
Not sure how that would look. Calculate ten times too high, convert to string and then insert the decimal point textually?

Reply:
Well, instead of something like:

if (x < 3600) return std::to_string(x/60) + " minutes";

Something like:

// top of the function:
std:stringstream ss; ss << std::fixed << std::setprecision(1);

then for each case:
(if x < 3600) { ss << (x/60.f); return ss.str() + " minutes";

Might need fixups, I did not compile this.

Reply:
Thanks, implemented as suggested.